01/components/ClockDate.js

```diff
@@ -9,10 +9,12 @@ const ClockDate = props => {
         return day.toString().padStart(2, '0');
     }
 
-    const { date } = props;
-    const year = date.getFullYear();
-    const month = getCorrectMonthFormat(date.getMonth());
-    const day = getCorrectDayFormat(date.getDate());
+    const {
+        date: { getDate, getFullYear, getMonth },
+    } = props;
```

Comment:
Gdy robisz destrukturyzację w ten sposób to `this` przy wywołaniu nie wskazuje już na `date` i jest błąd.
Powinno zostać `const { date } = props` i potem `date.getFullYear()`